datalake/src/main/java/com/sequenceiq/datalake/flow/create/SdxCreateActions.java

```diff
@@ -75,6 +85,11 @@ protected SdxContext createFlowContext(FlowParameters flowParameters, StateConte
 
             @Override
             protected void doExecute(SdxContext context, SdxEvent payload, Map<Object, Object> variables) throws Exception {
+                // When SDX is created as part of re-size flow chain, SDX in payload will not have ID.
+                if (!Strings.isNullOrEmpty(payload.getSdxName())) {
+                    SdxCluster sdxCluster = sdxService.getByNameInAccount(context.getUserId(), payload.getSdxName());
+                    context.setSdxId(sdxCluster.getId());
+                }
```

Comment:
what is the reason for this? based on description, this should be a new datalake in database and it should have id

Reply:
SdxCluster entity is created in SdxService.java but it is not persisted into the database immediately as there is a data lake already existing for the environment. It is persisted as part of detaching flow.

As it is not persisted at the flow chain creation, the payload(SdxCluster) here does not have the ID in it.

Reply:
I see, it makes sense, thanks for the explanation!